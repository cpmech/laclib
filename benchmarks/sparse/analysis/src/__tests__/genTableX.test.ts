import { genTableX } from '../genTableX';
import { IReport, IReportSet, zeroReport } from '../types';
import { writeHtml } from '../genHtml';

describe('genTableX', () => {
  const ro: IReport[] = [zeroReport(), zeroReport(), zeroReport(), zeroReport()];
  const ri: IReport[] = [zeroReport(), zeroReport(), zeroReport(), zeroReport()];

  ro[0].StepAnalyze.ElapsedTimeString = '1.1';
  ro[1].StepAnalyze.ElapsedTimeString = '1.2';
  ro[2].StepAnalyze.ElapsedTimeString = '1.3';
  ro[3].StepAnalyze.ElapsedTimeString = '1.4';

  ro[0].StepFactorize.ElapsedTimeString = '2.1';
  ro[1].StepFactorize.ElapsedTimeString = '2.2';
  ro[2].StepFactorize.ElapsedTimeString = '2.3';
  ro[3].StepFactorize.ElapsedTimeString = '2.4';

  ro[0].StepSolve.ElapsedTimeString = '3.1';
  ro[1].StepSolve.ElapsedTimeString = '3.2';
  ro[2].StepSolve.ElapsedTimeString = '3.3';
  ro[3].StepSolve.ElapsedTimeString = '3.4';

  ro[0].TimeSolverString = '4.1';
  ro[1].TimeSolverString = '4.2';
  ro[2].TimeSolverString = '4.3';
  ro[3].TimeSolverString = '4.4';

  ro[0].Stats.NormInfAx = 11.11;
  ro[1].Stats.NormInfAx = 22.22;
  ro[2].Stats.NormInfAx = 33.33;
  ro[3].Stats.NormInfAx = 44.44;

  ro[0].Stats.RelativeError = 0.11;
  ro[1].Stats.RelativeError = 0.22;
  ro[2].Stats.RelativeError = 0.33;
  ro[3].Stats.RelativeError = 0.44;

  ri[0].StepAnalyze.ElapsedTimeString = '11';
  ri[1].StepAnalyze.ElapsedTimeString = '12';
  ri[2].StepAnalyze.ElapsedTimeString = '13';
  ri[3].StepAnalyze.ElapsedTimeString = '14';

  ri[0].StepFactorize.ElapsedTimeString = '21';
  ri[1].StepFactorize.ElapsedTimeString = '22';
  ri[2].StepFactorize.ElapsedTimeString = '23';
  ri[3].StepFactorize.ElapsedTimeString = '24';

  ri[0].StepSolve.ElapsedTimeString = '31';
  ri[1].StepSolve.ElapsedTimeString = '32';
  ri[2].StepSolve.ElapsedTimeString = '33';
  ri[3].StepSolve.ElapsedTimeString = '34';

  ri[0].TimeSolverString = '41';
  ri[1].TimeSolverString = '42';
  ri[2].TimeSolverString = '43';
  ri[3].TimeSolverString = '44';

  ri[0].Stats.NormInfAx = 101;
  ri[1].Stats.NormInfAx = 202;
  ri[2].Stats.NormInfAx = 303;
  ri[3].Stats.NormInfAx = 404;

  ri[0].Stats.RelativeError = 0.01;
  ri[1].Stats.RelativeError = 0.02;
  ri[2].Stats.RelativeError = 0.03;
  ri[3].Stats.RelativeError = 0.04;

  it('generates default table', () => {
    const reports: IReportSet = {
      open: ro,
      intel: ri,
    };
    const table = genTableX('mytableX', reports);
    expect(table).toMatchSnapshot();
    const filepath = '/tmp/laclib/table.html';
    writeHtml(filepath, 'laclib table', table);
    console.log(`file <${filepath}> written`);
  });
});
