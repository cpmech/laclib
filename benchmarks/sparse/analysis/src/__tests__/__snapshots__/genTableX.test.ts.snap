// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genTableX generates default table 1`] = `
"<table style=\\"border: 1px solid #979797; border-collapse: collapse;\\">
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">mytableX</td>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">MPI np 1</th>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">MPI np 2</th>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">MPI np 3</th>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">MPI np 4</th>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"2\\">Analyze</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.1</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.2</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.3</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.4</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">11</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">12</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">13</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">14</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"2\\">Factorize</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2.1</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2.2</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2.3</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2.4</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">21</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">22</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">23</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">24</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"2\\">Solve</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3.1</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3.2</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3.3</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3.4</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">31</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">32</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">33</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">34</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"2\\">Total</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">4.1</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">4.2</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">4.3</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">4.4</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">41</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">42</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">43</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">44</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"2\\">Linf(A*x)</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">11.11</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">22.22</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">33.33</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">44.44</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">101</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">202</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">303</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">404</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"2\\">Rel Error</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.10e-1</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2.20e-1</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3.30e-1</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">4.40e-1</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.00e-2</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2.00e-2</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3.00e-2</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">4.00e-2</td>
  </tr>
</table>
"
`;
